app/controllers/application_controller.rb

```diff
@@ -8,9 +8,16 @@ class ApplicationController < ActionController::Base
   before_action :set_user
   before_action :configure_permitted_parameters, if: :devise_controller?
   before_action :check_authorization
+  before_action :check_max_session_time
 
 protected
 
+  def check_max_session_time
+    return false if current_user.nil?
+    time_out = Rails.application.secrets.session_expiry_in_minutes.nil? ? 60 : Rails.application.secrets.session_expiry_in_minutes.to_i
+    sign_out_user if Time.parse(current_user.current_sign_in_at) + time_out.minutes < Time.now
```

Comment:
being picky, but can we say `timeout` instead of `time_out` and `session_start_time` instead of `current_sign_in_at`?